tests/test_deserialization.py

```diff
@@ -422,6 +422,7 @@ def test_field_toggle_show_invalid_value_in_error_message(self):
         [
             "not-a-datetime",
             42,
+            0,
```

Comment:
thought(non-blocking): I believe this will also make False a valid value which isn't totally intuitive. But I'm not sure there's a way around that

Reply:
I added a condition to reject boolean inputs, copied from numeric fields above.

Reply:
I would be tempted to inherit `Float` to match `datetime.fromtimestamp()`'s type signature and avoid duplicating that logic. That would also allow removing `value = float(value)` from `utils.from_timestamp()` which is equally guilty for the unexpected bool to int promotion.

Reply:
On second thought inheritance wouldn't really work with the other date formats.

I would push this logic down to `utils.from_timestamp()` since all of the other date time `utils.from_*` properly error on bools already.

Reply:
https://github.com/marshmallow-code/marshmallow/commit/55cfa193068421906681894479342e2415c751df

Reply:
Right. I added your commit to the PR.